File: backend/app/models/parse_job.py
from datetime import datetime, timezone
from enum import StrEnum

from pydantic import Field

from app.models.lesson_document import ApiModel, LessonDocument


class JobStatus(StrEnum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class ErrorDetail(ApiModel):
    code: str
    message: str


class ParseJob(ApiModel):
    job_id: str = Field(alias="jobId")
    status: JobStatus
    source_file_name: str = Field(alias="sourceFileName")
    created_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc), alias="createdAt"
    )
    document: LessonDocument | None = None
    error: ErrorDetail | None = None

File: backend-java/src/main/java/com/lessonweb/lesson/model/job/ParseJob.java
package com.lessonweb.lesson.model.job;

import com.lessonweb.lesson.model.lesson.LessonDocument;

import java.time.Instant;

public record ParseJob(
        String jobId,
        JobStatus status,
        String sourceFileName,
        Instant createdAt,
        LessonDocument document,
        ErrorDetail error
) {
}
